### ISSUE TITLE
[bug] script relies on /usr/local/bin/facter but it does not always exist

### ISSUE THREAD

**Issue by GeoffWilliams** [albatrossflavour/puppet_os_patching#56]
**Overview**
`facter` is sometimes but not always present at `/usr/local/bin/facter`

**Impact**
Script failure
```
Error: /Stage[main]/Os_patching/Exec[os_patching::exec::fact]: Failed to call refresh: '/usr/local/bin/os_patching_fact_generation.sh' returned 1 instead of one of [0]
Error: /Stage[main]/Os_patching/Exec[os_patching::exec::fact]: '/usr/local/bin/os_patching_fact_generation.sh' returned 1 instead of one of [0]
```

**workaround**
ln -s /opt/puppetlabs/bin/facter /usr/local/bin/facter

**Longer term fix**
Hardcode /opt/puppetlabs/bin/facter ? :/ IDK ... Puppet doesn't make a fact available telling us where its installed stuff and PATH is unreliable

**Comment by albatrossflavour** [comment]
@GeoffWilliams and I talked on Slack and have decided to use the value from `facter path` as a default path set through the PATH variable.  This will allow us to remove the hard coded values from the `facter` calls.  Will update this on a feature branch and test.

**Comment by albatrossflavour** [comment]
Resolved in PR #62
